tests/services/test_links.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Test links."""
+
+import pytest
+from flask_principal import Identity, Need, UserNeed
+
+from invenio_records_resources.services.record import RecordService
+
+
+@pytest.fixture()
+def identity_simple():
+    """Simple identity fixture."""
+    i = Identity(1)
+    i.provides.add(UserNeed(1))
+    i.provides.add(Need(method='system_role', value='any_user'))
+    return i
+
+
+@pytest.fixture()
+def identity_no_need():
+    """Simple identity fixture."""
+    i = Identity(1)
+    # i.provides.add(UserNeed(1))
+    # i.provides.add(Need(method='system_role', value='any_user'))
+    return i
+
+
+def test_record_links(app, identity_simple, input_service_data, es):
+    record_service = RecordService()
+    record_unit = record_service.create(input_service_data, identity_simple)
+    pid_value = record_unit.id
+
+    links = record_service.linker.links(
```

Comment:
knowledge check: Since the links comes back from *creation* (in the unit). Why is the linker called again? My point being the test is:
- Creating a record (that comes with links)
- Creating the links again (in the same fashion)
- Comparing the second part to the expected.

I understand this is testing the `linker`, so the assertion is correct. However, for completeness we could check also the links are the same that came with the record? (though that might be a test in the service?)

``` python
assert expected_links == links == record_unit.links
```

It's okay, is just the test is not fully clear to me.

Reply:
Ah! I was also wondering at it this morning... and then I remembered. I will add the below to make it clearer what I was going for.

```
    # NOTE: We are testing linker.links() as opposed to record_unit.links
    #       because the former is used in all RecordService methods (not just
    #       create)
```

Essentially, it's to make the test unbiased to the `create()` method in particular, by testing the underlying method that is used in all RecordService methods. Then invenio-rdm-records' tests check all the ramifications.